src/pages/ServiceList/FiltersAndSorts.ts

```diff
@@ -54,18 +55,22 @@ export namespace ServiceListFilters {
       title: 'Health',
       isNumeric: false,
       param: 'he',
-      compare: (a: ServiceItemHealth, b: ServiceItemHealth) => {
-        const statusForA = a.health.getGlobalStatus();
-        const statusForB = b.health.getGlobalStatus();
+      compare: (a: ServiceListItem, b: ServiceListItem) => {
+        if (isServiceListItemHealth(a) && isServiceListItemHealth(b)) {
+          const statusForA = a.health.getGlobalStatus();
+          const statusForB = b.health.getGlobalStatus();
+
+          if (statusForA.priority === statusForB.priority) {
+            // If both services have same health status, use error rate to determine order.
+            const ratioA = getRequestErrorsStatus(a.health.requests.errorRatio).value;
+            const ratioB = getRequestErrorsStatus(b.health.requests.errorRatio).value;
+            return ratioA === ratioB ? a.name.localeCompare(b.name) : ratioB - ratioA;
+          }
 
-        if (statusForA.priority === statusForB.priority) {
-          // If both services have same health status, use error rate to determine order.
-          const ratioA = getRequestErrorsStatus(a.health.requests.errorRatio).value;
-          const ratioB = getRequestErrorsStatus(b.health.requests.errorRatio).value;
-          return ratioA === ratioB ? a.name.localeCompare(b.name) : ratioB - ratioA;
+          return statusForB.priority - statusForA.priority;
         }
 
-        return statusForB.priority - statusForA.priority;
+        return 0;
```

Comment:
Should we throw an error here? we are comparing two different things and returning 0 means that they are equal.

Reply:
I agree, but there is the same behavior on the other files, so I just made them consistent. Let's see what others think @mtho11 @israel-hdez @jotak

Reply:
Ah, OK.

Reply:
I'll have a closer look tomorrow but if I remember correctly this comparison method should only be called with items that have associated health. I guess the problem is that it may not be enforced by types?

Reply:
What @jotak is saying seems to be true, because the code that is below (which you also changed in this PR):

https://github.com/kiali/kiali-ui/blob/4cfcea6865ca355966fa1605a75c3a2fae445e86/src/pages/ServiceList/FiltersAndSorts.ts#L137-L146

is calling the `compare` function once all healths have been collected and the "augmented" object is ready. So, this `compare` should receive `WithHealth<ServiceItemList>` arguments.

Code is similar for Apps and Workloads case. Probably, to make things consistent, the other `{App,Workload}List/FilterAndSort.ts` are the ones that need the change to the type of its arguments.

Reply:
@israel-hdez @jotak if that's true, why does the current `compare` impl check if the item has health or not? I imagine it was changed somewhere else and became always true, so I think I'll refactor those functions and assume that it always has health data.

Reply:
@cfcosta I can't be 100% sure but it was maybe some zealous defensive check, that we can get rid of.